src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -101,23 +120,25 @@ trait RegularSync extends BlockBroadcast {
     bestPeer match {
       case Some(peer) =>
         val blockNumber = appStateStorage.getBestBlockNumber()
-        val request = GetBlockHeaders(Left(blockNumber + 1), blockHeadersPerRequest, skip = 0, reverse = false)
-        waitingForActor = Some(context.actorOf(
-          SyncBlockHeadersRequestHandler.props(peer, peerResponseTimeout, etcPeerManager, peerEventBus, request, resolveBranches = false)))
+        requestBlockHeaders(peer, GetBlockHeaders(Left(blockNumber + 1), blockHeadersPerRequest, skip = 0, reverse = false))
+        resolvingBranches = false
+
       case None =>
         log.debug("No peers to download from")
         scheduleResume()
     }
   }
 
-  private def handleBlockBranchResolution(peer: Peer, message: Seq[BlockHeader]) =
+  private def handleBlockBranchResolution(peer: Peer, message: Seq[BlockHeader]) = {
+    //todo limit max branch depth? [EC-248]
```

Comment:
this was implemented in https://github.com/input-output-hk/mantis/pull/286
bad merge?